command/agent/agent.go

```diff
@@ -127,28 +130,64 @@ func Create(config *Config, logOutput io.Writer) (*Agent, error) {
 		logOutput = os.Stderr
 	}
 
-	// Validate the config
-	if config.Datacenter == "" {
-		return nil, fmt.Errorf("Must configure a Datacenter")
-	}
-	if config.DataDir == "" && !config.DevMode {
-		return nil, fmt.Errorf("Must configure a DataDir")
-	}
-
-	// Try to get an advertise address
+	var bindIPs []string
 	if config.AdvertiseAddr != "" {
-		if ip := net.ParseIP(config.AdvertiseAddr); ip == nil {
-			return nil, fmt.Errorf("Failed to parse advertise address: %v", config.AdvertiseAddr)
+		// Parse config.AdvertiseAddr to resolve an IP address
+		out, err := template.Parse(config.AdvertiseAddr)
```

Comment:
I was prepping the code for multiple listeners, but backed out of that w/ the most recent push.  It will be easy enough to update and support when we're ready, however.